### PR TITLE
This PR: Add in UsesTraits trait.

### PR DESCRIPTION
**Pull request by kayladnls**

### CODE REVIEW

File: test/PHPUnit/Assertion/UsesTraitsTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    public function testGetsClassParentTraits()
+    {
+        $expected = [BuildsMockTrait::class];
+
+        $this->assertUsesTraits($expected, HasTraitClass::class);
+    }
+}
+
+class HasTraitClass
```
Comment (localheinz) [inline test/PHPUnit/Assertion/UsesTraitsTest.php]: @kayladnls 

Only one class per file, please.

File: test/PHPUnit/Assertion/UsesTraitsTest.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        $expected = [BuildsMockTrait::class];
+
+        $this->assertUsesTraits($expected, HasTraitClass::class);
+    }
+
+    public function testGetsClassParentTraits()
+    {
+        $expected = [BuildsMockTrait::class];
+
+        $this->assertUsesTraits($expected, HasTraitClass::class);
```
Comment (localheinz) [inline test/PHPUnit/Assertion/UsesTraitsTest.php]: @kayladnls 

There's no apparent difference between the two test cases.

**Comment by localheinz** [comment]
@kayladnls 

Generally, this assertion is probably not a good idea - because we're probably more concerned that a unit has a certain behavior, right? 

It made sense for untestable legacy code (in the sense of “The way of Testivus”), but maybe it would be better to mix in traits into tests, asserting that behavior?

What do you think?

**Comment by kayladnls** [comment]
I would prefer to test the trait functionality directly. But this was used everywhere, so I added it. 

If it's not needed I'll close it.

**Pull request closed without merging**